app/com/linkedin/drelephant/mapreduce/heuristics/DistributedCacheLimitHeuristic.java

```diff
@@ -0,0 +1,148 @@
+        logger.warn("Mismatch in the number of files and their corresponding sizes for " + MAPREDUCE_JOB_CACHE_FILES);
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_FILES, Integer.toString(cacheFilesArray.length));
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_FILES_FILESIZES, Integer.toString(cacheFileSizesArray.length));
+        return result;
+      }
+
+      Map<String, String> cacheFileToSizeMap = new HashMap<String, String>();
+      for (int i = 0; i < cacheFilesArray.length; i++) {
+        cacheFileToSizeMap.put(cacheFilesArray[i], cacheFileSizesArray[i]);
+      }
+
+      if (checkFileSizeLimit(result, cacheFileToSizeMap)) {
+        result.setSeverity(Severity.MODERATE);
+        logger.warn("Cache file size was found to be exceeding the max limit of " + distributedCacheFileSizeLimit
+            + " bytes for job " + jobName);
+      }
+    }
+
+    String archiveCacheFiles = jobConf.getProperty(MAPREDUCE_JOB_CACHE_ARCHIVES, null);
+    String archiveCacheFileSizes = jobConf.getProperty(MAPREDUCE_JOB_CACHE_ARCHIVES_FILESIZES, null);
+
+    if (archiveCacheFiles != null && archiveCacheFileSizes != null) {
+
+      if (result == null) {
+        result =
+            new HeuristicResult(_heuristicConfData.getClassName(), _heuristicConfData.getHeuristicName(), Severity.NONE,
+                0);
+      }
+
+      String[] archiveCacheFilesArray = archiveCacheFiles.split(",");
+      String[] archiveCacheFileSizesArray = archiveCacheFileSizes.split(",");
+
+      if (archiveCacheFilesArray.length != archiveCacheFileSizesArray.length) {
+        result.setSeverity(Severity.MODERATE);
+        logger
+            .warn("Mismatch in the number of files and their corresponding sizes for " + MAPREDUCE_JOB_CACHE_ARCHIVES);
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_ARCHIVES, Integer.toString(archiveCacheFilesArray.length));
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_ARCHIVES_FILESIZES,
+            Integer.toString(archiveCacheFileSizesArray.length));
+        return result;
+      }
+
+      Map<String, String> archiveCacheFileToSizeMap = new HashMap<String, String>();
+      for (int i = 0; i < archiveCacheFilesArray.length; i++) {
+        archiveCacheFileToSizeMap.put(archiveCacheFilesArray[i], archiveCacheFileSizesArray[i]);
+      }
+
+      if (checkFileSizeLimit(result, archiveCacheFileToSizeMap)) {
+        result.setSeverity(Severity.MODERATE);
+        logger.warn(
+            "Cache archive file size was found to be exceeding the max limit of " + distributedCacheFileSizeLimit
+                + " bytes for job " + jobName);
+      }
+    }
+
+    return result;
+  }
+
+  private boolean checkFileSizeLimit(HeuristicResult result, Map<String, String> cacheFileToSizeMap) {
+    boolean limitViolated = false;
+    for (String file : cacheFileToSizeMap.keySet()) {
+      long size = Long.parseLong(cacheFileToSizeMap.get(file));
+      if (size > distributedCacheFileSizeLimit) {
```

Comment:
You are check each jar size limit to be > 500MB. Shouldn't it be the sum of all sizes > 500 MB?